### PR TITLE
describe what constitutes a minor revision to nbformat

### LINKED ISSUE #6519: formailize allowed backward-compatible nbformat changes
Now that #6045 introduces notebook validation, we should define what sort of backward-compatible changes are allowed in the notebook format that qualify for a minor revision. In 3.x-series, we defined it loosely as new properties that were safely ignored (metadata was added in various places, under this rule). The current validation supports this by skipping any `additionalProperties: false` validations for notebooks from the future. Other possible backward-compatible changes we could allow:

- new output types
- new cell types

These would need to be handled in the code (mainly javascript) by loading the unrecognized output or cell with a simple placeholder. It need not be editable, just indicate that there's something there that the UI doesn't understand.

We don't need to have this all done before merging #6045, but we should have a pretty formal idea of what sort of extensions we allow before our next backward-incompatible nbformat change (hopefully a very long time from now).

### PR DESCRIPTION
**Pull request by minrk**
and fix the stream output spec, which had fallen out of date

closes #6519

### CODE REVIEW

File: docs/source/notebook/nbformat.rst [hunk 1]
```diff
@@ -291,6 +291,22 @@ regardless of format.
       "source" : ["some nbformat mime-type data"]
     }
 
+Backwared-compatible changes
```
Comment (jasongrout) [inline docs/source/notebook/nbformat.rst]: "Backward"

**Comment by jasongrout** [comment]
How did the changes to IPython/html/templates/tree.html end up in this PR?

File: docs/source/notebook/nbformat.rst [hunk 2]
```diff
@@ -291,6 +291,22 @@ regardless of format.
...
+============================
+
+The notebook format is an evolving format. When backward-compatible changes are made,
+the notebook format minor version is incremented. When backward-incompatible changes are made,
+the major version is incremented.
+
+As of nbformat 4.x, backward-compatible changes include:
+
+- new fields in any cell or output or metadata
```
Comment (takluyver) [inline docs/source/notebook/nbformat.rst]: Specifically in these places, or in any dictionary? E.g. if we add new keys to the top level notebook object, is that backwards compatible?

Comment (minrk) [inline docs/source/notebook/nbformat.rst]: Any dictionary, I guess.

File: docs/source/notebook/nbformat.rst (hunk 1 above)
Comment (minrk) [inline docs/source/notebook/nbformat.rst]: fixed

File: docs/source/notebook/nbformat.rst (hunk 2 above)
Comment (minrk) [inline docs/source/notebook/nbformat.rst]: added.

**Pull request merged**